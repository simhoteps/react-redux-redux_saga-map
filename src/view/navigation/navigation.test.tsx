import { render, screen } from "@testing-library/react";
import { Size, useWindowSize } from "core/custom-hooks/use_window_size";
import { BrowserRouter } from "react-router-dom";
import Navigation from "./navigation";
import Home from "../home/home";

test("Navigation Container renders correctly", async () => {
  render(
    <BrowserRouter>
      <Navigation />
    </BrowserRouter>
  );

  const containerEl = screen.getByTestId("navigationContainer");
  expect(containerEl).toBeInTheDocument();

  const contentEl = screen.getByTestId("navigationContent") as HTMLImageElement;
  expect(contentEl).toHaveStyle("overflow: hidden");
});

it("renders without crashing", async () => {
  const wrapper = render(
    <BrowserRouter>
      <Navigation />
    </BrowserRouter>
  );
  expect(wrapper).toMatchSnapshot();
  wrapper.unmount();
});
