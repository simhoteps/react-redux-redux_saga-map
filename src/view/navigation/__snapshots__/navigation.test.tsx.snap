// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="navigationContainer"
        style="display: flex; overflow-x: hidden; background-color: rgba(54, 84, 98, 0.063);"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1naxzgn-MuiPaper-root-MuiAppBar-root"
          data-testid="headerAppBar"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-l8v6ha-MuiToolbar-root"
          >
            <div
              class="MuiBox-root css-kg9p2d"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-14twn8l-MuiTypography-root"
                data-testid="titleName"
              >
                Coronavirus (COVID-19) Dashboard
              </p>
              <div
                class="MuiBox-root css-1mud55e"
              />
            </div>
          </div>
        </header>
        <div
          class="MuiBox-root css-81f0u5"
          data-testid="navigationContent"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="navigationContainer"
      style="display: flex; overflow-x: hidden; background-color: rgba(54, 84, 98, 0.063);"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1naxzgn-MuiPaper-root-MuiAppBar-root"
        data-testid="headerAppBar"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-l8v6ha-MuiToolbar-root"
        >
          <div
            class="MuiBox-root css-kg9p2d"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-14twn8l-MuiTypography-root"
              data-testid="titleName"
            >
              Coronavirus (COVID-19) Dashboard
            </p>
            <div
              class="MuiBox-root css-1mud55e"
            />
          </div>
        </div>
      </header>
      <div
        class="MuiBox-root css-81f0u5"
        data-testid="navigationContent"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
